TokenRateTest.php

```diff
@@ -0,0 +1,50 @@
+<?
+
+namespace iFixit\TokenBucket;
+
+error_reporting(E_ALL);
+
+use \DateInterval;
+
+use PHPUnit_Framework_TestCase;
+
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'TokenRate.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'TokenBucket.php';
+
+
+class TokenRateTest extends PHPUnit_Framework_TestCase {
+   public function testRate() {
+      $rate = new TokenRate(1, new DateInterval("PT01S"));
+      $this->assertSame(1.0, $rate->getRate());
+   }
+
+   public function testLargerRate() {
+      $rate = new TokenRate(10, new DateInterval("PT10S"));
+      $this->assertSame(1.0, $rate->getRate());
+   }
+
+   public function testZeroRate() {
+      $rate = new TokenRate(0, new DateInterval("PT10S"));
+      $this->assertSame(0, $rate->getRate());
+   }
+
+   /**
+    * I'm not really sure what this means for token buckets, but it's still a
+    * rate.
+    */
+   public function testNegativeTokenRate() {
+      $rate = new TokenRate(-5, new DateInterval("PT10S"));
```

Comment:
@danielbeardsley I would argue giving just a seconds parameter makes it even more inconvenient to specify time. Sure the `PT` part of `PT10S` could be confusing, but seeing `new TokenRate(5, 1800)` is harder to see what's going on compared to `new TokenRate(5, new DateInterval("PT30M"))`